tst/EnergyPlus/unit/UnitHeater.unit.cc

```diff
@@ -1372,3 +1374,1087 @@ TEST_F( EnergyPlusFixture, UnitHeater_SimUnitHeaterTest ) {
+		"    AvailabilityManager:LowTemperatureTurnOff,  !- Availability Manager 1 Object Type",
+		"    Chilled Water Loop Availability Low Temp Off;  !- Availability Manager 1 Name",
+
+		"AvailabilityManager:LowTemperatureTurnOff,",
+		"    Chilled Water Loop Availability Low Temp Off,  !- Name",
+		"    Chilled Water Loop Outside Air Sensor,  !- Sensor Node Name",
+		"    7.22;                    !- Temperature {C}",
+
+		"OutdoorAir:Node,",
+		"    Chilled Water Loop Outside Air Sensor,  !- Name",
+		"    -1;                      !- Height Above Ground {m}",
+	} );
+	ASSERT_FALSE( process_idf( idf_objects ) );
+
+	OutputProcessor::TimeValue.allocate( 2 );
+	DataGlobals::DDOnlySimulation = true;
+
+	ManageSimulation();
+
+	EXPECT_EQ( ZoneEquipList( 1 ).NumOfEquipTypes, 2 );
+	// first priority zone equipment is zone ADU
+	EXPECT_EQ( ZoneEquipmentManager::PrioritySimOrder( 1 ).EquipType, "ZONEHVAC:AIRDISTRIBUTIONUNIT" );
+	EXPECT_EQ( ZoneEquipmentManager::PrioritySimOrder( 1 ).EquipName, "MAIN ZONE ATU" );
+	EXPECT_EQ( HeatingCoils::HeatingCoil( 1 ).Name, "MAIN ZONE REHEAT COIL" );
+	// second priority zone equipment is unit heater
+	EXPECT_EQ( ZoneEquipmentManager::PrioritySimOrder( 2 ).EquipType, "ZONEHVAC:UNITHEATER" );
+	EXPECT_EQ( ZoneEquipmentManager::PrioritySimOrder( 2 ).EquipName, "UNITHEATER" );
+	EXPECT_EQ( HeatingCoils::HeatingCoil( 2 ).Name, "UNITHEATER_ELECTRICHEATER" );
+	// check the reheat coil output
+	EXPECT_NEAR( HeatingCoils::HeatingCoil( 1 ).HeatingCoilRate, 7015.5, 1.0 );
+	// check the unit heater heating coil output
+	EXPECT_EQ( HeatingCoils::HeatingCoil( 2 ).HeatingCoilRate, 0.0 );
+
+	// re-set the hour of the day
+	DataGlobals::TimeStep = 1;
+	DataGlobals::HourOfDay = 24;
+	DataGlobals::CurrentTime = 24.0;
+	// set zone air node condition
+	Node( ZoneEquipConfig( 1 ).ZoneNode ).Temp = 20.0;
+	Node( ZoneEquipConfig( 1 ).ZoneNode ).HumRat = 0.005;
+	Node( ZoneEquipConfig( 1 ).ZoneNode ).Enthalpy = Psychrometrics::PsyHFnTdbW( Node( ZoneEquipConfig( 1 ).ZoneNode ).Temp, Node( ZoneEquipConfig( 1 ).ZoneNode ).HumRat );
+	// set the zone loads
+	ZoneSysEnergyDemand( 1 ).TotalOutputRequired = 0.0;
+	ZoneSysEnergyDemand( 1 ).OutputRequiredToHeatingSP = 15000.0;
+	ZoneSysEnergyDemand( 1 ).OutputRequiredToCoolingSP = 20000.0;
+	// local variables
+	bool SimZoneEquipment = true;
+	bool SimAirLoops = true;
+	bool FirstHVACIteration = false;
+	// re-simulate the zone HVAC equipment per the priority order
+	ZoneEquipmentManager::ManageZoneEquipment( FirstHVACIteration, SimZoneEquipment, SimAirLoops );
+	// check the reheat coil nominal capacity
+	EXPECT_NEAR( HeatingCoils::HeatingCoil( 1 ).NominalCapacity, 17542.3, 1.0 );
+	// check the reheat coil outputis the full capacity
+	EXPECT_NEAR( HeatingCoils::HeatingCoil( 1 ).HeatingCoilRate, 17542.3, 1.0 );
+	// check the unit heater heating coil is handling the remaning load
+	EXPECT_NEAR( HeatingCoils::HeatingCoil( 2 ).HeatingCoilRate, 223.7, 1.0 );
+
```

Comment:
@Nigusse can you check RemaingingOutputRequired to see if it is 0? Or RemaingingOutputRequired"to something".